src/shared.rs

```diff
@@ -14,6 +14,16 @@ pub fn get_argument_names(
     args.keys().map(String::from).collect()
 }
 
+#[allow(clippy::implicit_hasher)]
+pub fn get_optional_argument_names(
+    args: &std::collections::BTreeMap<
+        String,
+        flux::semantic::types::Argument<MonoType>,
```

Comment:
Nit: Could just make `get_argument_names` above generic over the values. ` &std::collections::BTreeMap<String, T>`